src/utils/index.ts

```diff
@@ -0,0 +1,56 @@
+import type { Data, Inputs, AttributeVal, HtmlAttributes } from '../types';
+
+function formatUrl(url: string, params: string[], args: Inputs) {
```

Comment:
Nit: do you want to make `params` and `args` optional since they may be undefined? (will also allow you to remove the "!" lower down)

Reply:
Good call, updated